### PR TITLE
Added removeOutgoingStore function.

### PR DESCRIPTION
**Pull request by redboltz**
I'd like to have removing a message from outgoing store functionality. I connect a MQTT broker using MQTT.js. The broker have permission control functionality. That means a client can publish only if the client have permission. The permission is defined a pair of ClientId (or Username) and topic.
Permission management is out of scope of MQTT spec, but there are some brokers that support it.

When I publish QoS1 message to a topic that is not permitted, then the broker doesn't return puback because the message is not processed. Some brokers immediately disconnect, but other brokers keep connection and report errors using some special topic like `$SYS/error_reporpt` by publish back.  

I'm using a latter type broker. After the client received the error, I'd like to remove in-flight publish message. Because it still consuming packet id. In addition, when the client disconnect and connect again, the client will resend the message, and will get the same error.

So I'd like to remove the message.

I implement the function and tests. If it has some problem, I will improve my pull request.

### CODE REVIEW

File: lib/store.js [hunk 1]
```diff
@@ -83,6 +83,14 @@ Store.prototype.del = function (packet, cb) {
 }
 
 /**
+ * deletes a packet from the store by messageId
+ */
+Store.prototype.delByMessageId = function (messageId) {
+  delete this._inflights[messageId]
+  return this
+}
```
Comment (mcollina) [inline lib/store.js]: This new method is not needed, you can use https://github.com/redboltz/MQTT.js/blob/c35e0a615b44bb235fecffb6b2ae8bdee4556c5a/lib/store.js#L74 and call it with `{ messageId }`.

Comment (redboltz) [inline lib/store.js]: If I use `del`, and matches `messageId`, undefined `cb` seems to be called.

https://github.com/redboltz/MQTT.js/blob/c35e0a615b44bb235fecffb6b2ae8bdee4556c5a/lib/store.js#L77

Can I update `del` as follows?

Before
```JavaScript
    cb(null, packet)
```

After
```JavaScript
    if (cb) {
      cb(null, packet)
    }
```

**Comment by redboltz** [comment]
Thank you for the comment.

> Can you please update the docs as well?

Do you mean README.md? I will do that soon.

File: lib/store.js (hunk 1 above)
Comment (mcollina) [inline lib/store.js]: you can pass `nop` to  `del`.

**Comment by mcollina** [comment]
Yes, `README.md`.

File: lib/store.js (hunk 1 above)
Comment (redboltz) [inline lib/store.js]: Thanks, I updated it.

**Comment by redboltz** [comment]
I added README.md.

File: lib/store.js (hunk 1 above)
Comment (redboltz) [inline lib/store.js]: nodejs 7 and 8 reports an error on travis-ci. I updated `{ messageId }` to `{ messageId: mid}`, and changed the parameter name to `mid`.

File: lib/client.js [hunk 2]
```diff
@@ -597,6 +597,21 @@ MqttClient.prototype.end = function (force, cb) {
...
+ * removeOutgoingStore - remove a message in outgoing store
+ *
+ * @param {Number} mid - messageId to remove message
+ * @returns {MqttClient} this - for chaining
+ * @api public
+ *
+ * @example client.removeOutgoingStore(client.getLastMessageId());
+ */
+MqttClient.prototype.removeOutgoingStore = function (mid) {
+  delete this.outgoing[mid]
```
Comment (mcollina) [inline lib/client.js]: Maybe we should call the outgoing callback (if exists) with an error?

File: lib/client.js [hunk 3]
```diff
@@ -597,6 +597,21 @@ MqttClient.prototype.end = function (force, cb) {
...
 /**
+ * removeOutgoingStore - remove a message in outgoing store
+ *
+ * @param {Number} mid - messageId to remove message
+ * @returns {MqttClient} this - for chaining
+ * @api public
+ *
+ * @example client.removeOutgoingStore(client.getLastMessageId());
+ */
+MqttClient.prototype.removeOutgoingStore = function (mid) {
```
Comment (mcollina) [inline lib/client.js]: the name could be a bit more meaningful, something like `removeOutgoingMessage`.

File: lib/client.js (hunk 2 above)
Comment (redboltz) [inline lib/client.js]: I agree. It is better to inform the client the result. I updated the code, test, and document.

**Comment by redboltz** [comment]
Thank you very much!

**Pull request merged**